File: Byte&Buy.Backend/src/ByteBuy.Core/DTO/Public/Order/Rent/UserRentOrderLineResponse.cs
using ByteBuy.Core.DTO.Public.Money;
using ByteBuy.Core.DTO.Public.Order.Common;

namespace ByteBuy.Core.DTO.Public.Order.Rent;

public record UserRentOrderLineResponse : UserOrderLineResponse
{
    public MoneyDto PricePerDay { get; set; } = null!;
    public int RentalDays { get; set; }
}

File: Byte&Buy.AngularSPA/src/app/core/dto/order/rent/user-rent-order-line-response.ts
import { MoneyDto } from "../../common/money-dto";
import { UserOrderLineResponse } from "../common/user-order-line-response";

export interface UserRentOrderLineResponse extends UserOrderLineResponse {
    type: "rent";
    pricePerDay: MoneyDto;
    rentalDays: number;
}